util/chplenv/printchplbuilds.py

```diff
@@ -0,0 +1,443 @@
+# If 'all' is True then returns all environment variables, otherwise omits variables not
+# relevant to the current configuration.
+
+def GetConfig(all=False):
+    config = OrderedDict()
+    excludes = ['CHPL_HOME']
+    filters = ['no-tidy'] if all else ['tidy']
+    filters.append('only-path')
+    lines = printchplenv.printchplenv(['runtime'], print_filters=filters, 
+                                      print_format='simple').split('\n')
+    for line in lines:
+        if len(line) == 0:
+            continue
+        (var, value) = line.split('=', 1)
+        if var in excludes:
+            continue
+        config[var] = value
+        if var == 'CHPL_COMM' and value != "none":
+            # CHPL_COMM_DEBUG isn't returned by printchplenv
+            value = '+' if os.getenv('CHPL_COMM_DEBUG') else '-'
+            config['CHPL_COMM_DEBUG'] = value
+    return config
+
+# Returns a list of variables that differ between two configurations 
+
+def FindDiffVars(usedvars, a, b):
+    result = []
+    for var in usedvars:
+        value = a.get(var, 'NA')
+        if a.get(var, 'NA') != b.get(var, 'NA'):
+            result.append(var)
+    return result
+
+# print shell commands to change the environment variables from current to config
+
+def PrintShellCommands(current, config, shell="bash"):
+    global usedvars
+    if shell == "csh":
+        print("set noglob;")
+    for var in usedvars:
+        newValue = config.get(var, None)
+        oldValue = current.get(var, None)
+        if oldValue != newValue:
+            if newValue is None or newValue == '-':
+                if shell == "csh":
+                    print('unsetenv %s;' % (var))
+                elif shell == "bash":
+                    print('unset %s;' % (var))
+            else:
+                if newValue == '+':
+                    newValue = 1
+                if shell == "csh":
+                    print('setenv %s "%s";' %(var, newValue))
+                elif shell == "bash":
+                    print('export %s="%s";' %(var, newValue))
+    if shell == "csh":
+        print("unset noglob;")
+
+# Prints the configurations in columns. If a variable's value differs from the
+# current configuration it has a '*' suffix. "extras" are variables that shouldn't
+# get a prefix, e.g. MTIME.
+
+def Output(current, variables, extras, configs):
+    for var in variables + extras:
+        output = "%25s:" % var
+        for config in configs:
+            value = config.get(var, 'NA')
+            if var not in extras:
+                tag = "" if current.get(var, 'NA') == value else "*"
+                value += tag
+            if var == 'MTIME' and value != 'NA':
+                value = datetime.datetime.fromtimestamp(value).strftime("%b %d %H:%M")
+            output += " %-20s" % value
+        print(output)
+
+# For argument parsing
+@unique
+class Sort(Enum):
+    NEWEST =    1,
+    OLDEST =    2,
+    FEWEST =    3,
```

Comment:
I think the trailing comma here makes them tuples. Shouldn't actually cause an issue if you only ever compare by value but probably not intentional